packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts

```diff
@@ -189,7 +189,8 @@ class FirebaseAnalyticsJS {
       !eventName.length ||
       eventName.length > 40 ||
       eventName[0] === '_' ||
-      !eventName.match(/^[A-Za-z_]+$/) ||
+      eventName[0].match(/^\d$/) ||
+      !eventName.match(/^[A-Za-z_0-9]+$/) ||
```

Comment:
Perhaps we can provide a cleaner solution as per https://github.com/expo/expo/pull/8391/files ?

Reply:
Sure. Perhaps these two sections could also be combined into one method as they have the same logic minus the length.

Reply:
Feel free to reuse the regex! I think your PR is more complete, so I've closed mine